### PR TITLE
EDU-1412 Unlimited upload

### PR DESCRIPTION
**Pull request by ahelmberger**
Closes https://educandu.atlassian.net/browse/EDU-1412

### CODE REVIEW

File: src/domain/multipart-middleware.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    return requestSize && requestSize > EFFECTIVE_REQUEST_LIMIT_IN_BYTES && !allowUnrestricted
+      ? next(new BadRequest(`Upload limit exceeded: limit ${prettyBytes(EFFECTIVE_REQUEST_LIMIT_IN_BYTES)}, requested ${prettyBytes(requestSize)}.`))
+      : next();
+  };
+
+  const multipartParser = multer({
+    storage: multerS3({
+      s3: cdn.s3Client.client,
+      bucket: cdn.bucketName,
+      key: (_req, file, cb) => cb(null, `${getTemporaryUploadsPath()}/upload_${createUniqueStorageFileName(file.originalname)}`),
```
Comment (adrianaluchian) [inline src/domain/multipart-middleware.js:29]: Is the `upload_` prefix for the file name necessary for AWS' TTL?

Comment (ahelmberger) [reply src/domain/multipart-middleware.js:29]: Yes, that was the intention, in order to have the `__DIRMARKER__` excluded from automatic deletion.

File: src/server/media-library-controller.js [hunk 2]
```diff
@@ -125,15 +124,7 @@ class MediaLibraryController {
...
       needsPermission(permissions.CREATE_CONTENT),
-      uploadLimitExceededMiddleware(),
-      multipartParser.single('file'),
-      validateFile('file'),
-      (req, _res, next) => {
-        // Transform the body as if it were a REST call,
-        // so we can nicely validate it and work with it
-        req.body = JSON.parse(req.body.metadata || 'null') || {};
-        next();
-      },
```
Comment (adrianaluchian) [inline src/server/media-library-controller.js:136]: Nice that this is all gathered away in the middleware.

File: src/stores/cdn.js [hunk 3]
```diff
@@ -55,9 +55,15 @@ class Cdn {
   uploadObject(objectPath, filePath) {
     const metadata = this._getDefaultMetadata();
     const stream = fs.createReadStream(filePath);
-    const sanitizedObjectName = objectPath.replace(/\\/g, '/');
```
Comment (adrianaluchian) [inline src/stores/cdn.js:58]: I think we used to have some issues in S2 due to the `\` (it was considering the object name a path or something like that?). Is this sanitization not needed anymore?

Comment (ahelmberger) [reply src/stores/cdn.js:58]: I think it is a relic from elmu days, where we did not properly slugify all uploaded file names. Now we can be sure that no path separators or other URL-unsafe characters are in the file names.

File: src/domain/multipart-middleware.js (hunk 1 above)
Comment (adrianaluchian) [reply src/domain/multipart-middleware.js:29]: Got it 👍 Was wondering why the path/"folder" name was not enough.

**Review by adrianaluchian — APPROVED** [review]

**Pull request merged**